pkg/controller/build/build_controller.go

```diff
@@ -459,6 +468,25 @@ func (r *ReconcileBuild) validateBuildRunOwnerReferences(ctx context.Context, b
 	return nil
 }
 
+// validateSourceUrl returns error message if remote repository doesn't exist, otherwise, it will return empty string and nil
+func (r *ReconcileBuild) validateSourceUrl(ctx context.Context, b *build.Build, sourceUrl string) (string, error) {
+	prod, _ := zap.NewProduction()
+	logger := prod.Sugar()
+	switch b.GetAnnotations()[build.AnnotationBuildRefRepository] {
+	case "true":
+		if output, err := git.ListRemote(logger, sourceUrl); err != nil {
+			return output, err
+		} else {
+			return "", nil
+		}
+	case "", "false":
+		return "", nil
+	default:
+		ctxlog.Info(ctx, fmt.Sprintf("the annotation %s was not properly defined, supported values are true or false", build.AnnotationBuildRefRepository), namespace, b.Namespace, name, b.Name)
```

Comment:
Since the empty string is treated as `false`, I am wondering whether this should also hinted at in the info text and/or error message to avoid confusion.

Reply:
Good idea!

Reply:
Why are we treating "" as false here? Shouldn't that case fall through to the default, especially since the default says "supported values are true or false". If the annotation does not exist, that should also be treated as false/this check should be skipped.